unity-renderer/Assets/DCLPlugins/LoadingScreenPlugin/LoadingScreenPlugin.cs

```diff
@@ -8,17 +11,41 @@ namespace DCLPlugins.LoadingScreenPlugin
     /// </summary>
     public class LoadingScreenPlugin : IPlugin
     {
-        private readonly LoadingScreenController loadingScreenController;
+        private const string LOADING_SCREEN_ASSET = "_LoadingScreen";
+
         private readonly DataStoreRef<DataStore_LoadingScreen> dataStoreLoadingScreen;
+        private readonly IAddressableResourceProvider assetsProvider;
+        private readonly CancellationTokenSource cancellationTokenSource;
+
+        private LoadingScreenController loadingScreenController;
 
-        public LoadingScreenPlugin()
+        public LoadingScreenPlugin(IAddressableResourceProvider assetsProvider)
         {
+            this.assetsProvider = assetsProvider;
+
             dataStoreLoadingScreen.Ref.decoupledLoadingHUD.visible.Set(true);
-            loadingScreenController = new LoadingScreenController(LoadingScreenView.Create(), Environment.i.world.sceneController, Environment.i.world.state, NotificationsController.i,DataStore.i.player,
-                DataStore.i.common, dataStoreLoadingScreen.Ref, DataStore.i.realm);
+
+            cancellationTokenSource = new CancellationTokenSource();
+            CreateLoadingScreen(cancellationTokenSource.Token).Forget();
         }
 
-        public void Dispose() =>
+        private async UniTask CreateLoadingScreen(CancellationToken cancellationToken = default)
+        {
+            loadingScreenController = new LoadingScreenController(
+                await CreateLoadingScreenView(assetsProvider, cancellationToken),
```

Comment:
We should add a retry count, just in case any of the addresables request fail

Reply:
BTW it could be done here:
![image](https://user-images.githubusercontent.com/118179774/217483020-fa193210-c803-4269-a3b2-b7c29e68be57.png)

Reply:
So `0` means no retries, right?

Reply:
I think in this approach, Essential group will become dump of everything and can grow very fast. I even not sure that application will not function without loading screen 🤔 But could be. For now I will move LoadingScreen there